cloudify_types/cloudify_types/component/operations.py

```diff
@@ -66,16 +66,52 @@ def _get_desired_operation_input(key, args):
 def _get_client(kwargs):
     client_config = _get_desired_operation_input('client', kwargs)
     if client_config:
-        return CloudifyClient(**client_config)
+        client = CloudifyClient(**client_config)
+
+        # for determining an external client:
+        manager_ips = [mgr.private_ip for mgr in
+                       manager.get_rest_client().manager.get_managers()]
+        internal_hosts = ({'127.0.0.1', 'localhost'} | set(manager_ips))
+        host = {client.host} if type(client.host) == str \
```

Comment:
```suggestion
        host = {client.host} if isinstance(client.host, str) \
```
as a general rule, always prefer isinstance over `type()==`, because `type()==` ignores subclasses (breaking the liskov substitution principle). Doesn't matter in this case, and I'm aware this is just copied, but let's fix it while we're at it.